### PR TITLE
refactor(form components): use fields in form with component property

### PR DESCRIPTION
**Pull request by HendrikThePendric**
I would argue we should implement this approach for all the form components.

### CODE REVIEW

File: src/components/form/Select.js [hunk 1]
```diff
@@ -9,57 +8,47 @@ import { formInput, formOptions } from '../../utils/react/propTypes'
...
+                error={props.error}
+                touched={props.touched}
+            />
+        </FormControl>
+    </div>
 )
 
-StyledSelect.propTypes = {
+Select.propTypes = {
     classes: PropTypes.shape({
```
Comment (Mohammer5) [inline src/components/form/Select.js]: At first I added those as well, but then I realised that withStyles is like a HOC that will handle all the stuff, so I thought I just leave it out. There weren't any linting errors either.
I think this should go out, but just a preference, not a requested change

File: src/components/form/Select.js [hunk 2]
```diff
@@ -9,57 +8,47 @@ import { formInput, formOptions } from '../../utils/react/propTypes'
...
+                {props.placeholder}
+            </InputLabel>
+            <SelectField
+                input={props.input}
+                showEmptyOption={props.showEmptyOption}
+                options={props.options}
+            />
+            <ErrorText
+                showErrorText={props.showErrorText}
+                error={props.error}
```
Comment (Mohammer5) [inline src/components/form/Select.js]: `error` and `touched` aren't normal props anymore this way, they must be accessed through `meta`:
```error={props.meta.error}```

File: src/components/form/Select.js [hunk 3]
```diff
@@ -70,19 +59,9 @@ StyledSelect.propTypes = {
     touched: PropTypes.bool.isRequired,
```
Comment (Mohammer5) [inline src/components/form/Select.js]: `error` propType is missing. but as they're wrapped with the `meta` object, both `error` and `touched` propTypes need to be replaced with a meta propType. I guess we can outsource that to the same file where I put the formInput propType

File: src/components/form/Select.js (hunk 1 above)
Comment (HendrikThePendric) [reply src/components/form/Select.js]: I'm fine leaving them out. I think my linter complained, that's why I added em.

**Comment by Mohammer5** [comment]
Will be merged into another feature branch.

**Pull request closed without merging**